### PR TITLE
Removed profile from test metadata where not needed

### PR DESCRIPTION
**Pull request by matejak**
Historically, the profile had to be specified in test scenarios - oscap needs a profile that selects the rule to be able to scan for that rule using the `--rule` option. This became unnecessary when the `(alll)` virtual profile got introduced.

Nowadays, the `--profile` option is needed iff the rule is parametrized by an XCCDF variable that assigns a specific value to that variable. 

This PR removes the profile metadata from rules that have been identified as not parametrized using this thought: If a rule doesn't have `external` in their OVALs, or it doesn't have any `oval` directory at all, then the rule is not parametrized.

Here is the supportive bash code that was used to identify trees of test scenarios that have `profiles` metadata, but that don't need them.

```
test_trees_of_rules_with_profiles=$(grep -l profiles $(fd .pass.sh) $(fd .notapplicable.sh) $(fd .fail.sh) | sed -e 's|/tests/.*$|/tests|' | sort | uniq)
trees_with_external_vars=''
for tree in $test_trees_of_rules_with_profiles; do
    test -d "$tree/../oval" && grep -q external "$tree"/../oval/*.xml && trees_with_external_vars="$tree $trees_with_external_vars"
    grep -q '\<name:\s\+\(accounts_password\|mount_option_removable_partitions\|sebool\|sysctl\|yamlfile_value\)' "$tree"/../rule.yml && trees_with_external_vars="$tree $trees_with_external_vars"
done
trees_without_external_vars=
for tree in $test_trees_of_rules_with_profiles; do
    grep -q "\<$tree\>" <<< "$trees_with_external_vars" || trees_without_external_vars="$tree $trees_without_external_vars"
done
```

### CODE REVIEW

**Comment by jan-cerny** [comment]
> or it doesn't have any oval directory at all, then the rule is not parametrized.

Is it correct to assume that these rules aren't parametrized? They are often templated rules. I have seen that some templates reference an external variable. I haven't checked if they're tested, though.

> $(fd .pass.sh) $(fd .fail.sh)

What about the tests with expected result other than pass or fail, for example error?

**Comment by jan-cerny** [comment]
The following templates are referencing an external variable, so they might be worth checking.

template_OVAL_accounts_password
template_OVAL_mount_option_removable_partitions
template_OVAL_sebool
template_OVAL_sysctl
template_OVAL_yamlfile_value


We have noticed that if you remove a profile tag from all scenario for templated rules you can accidentally remove it even if it would be needed.

File: linux_os/guide/services/cron_and_at/service_atd_disabled/tests/service_disabled.pass.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 #
-# profiles = xccdf_org.ssgproject.content_profile_standard
+# packages = at
```
Comment (jan-cerny) [inline linux_os/guide/services/cron_and_at/service_atd_disabled/tests/service_disabled.pass.sh:3]: Should the `yum` command be removed, then?

Comment (matejak) [reply linux_os/guide/services/cron_and_at/service_atd_disabled/tests/service_disabled.pass.sh:3]: Not yet, there is no support for package metadata, but it is coming.

File: linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pwquality/accounts_password_pam_ocredit/tests/ospp.pass.sh [hunk 2]
```diff
@@ -1,5 +1,4 @@
 #!/bin/bash
 #
-# profiles = xccdf_org.ssgproject.content_profile_ospp
```
Comment (jan-cerny) [inline linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pwquality/accounts_password_pam_ocredit/tests/ospp.pass.sh:3]: There is variable `var_password_pam_ocredit.var` involved in this and for some profiles, this shouldn't pass, eg. rht-ccp profile requires ocredit to be set to -2.

File: linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_login_defs/tests/wrong_value.fail.sh [hunk 3]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash
 #
-# profiles = xccdf_org.ssgproject.content_profile_ospp
```
Comment (jan-cerny) [inline linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_login_defs/tests/wrong_value.fail.sh:3]: This depends on var_accounts_user_umask. In some profiles, it's set to 027 and in other profiles it's set to 077, so in some profiles this test should pass.

**Comment by matejak** [comment]
I have rebased the PR with following changes to the previous state:

- Package installation and package metadata was not touched at all.
- Profile metadata is kept if a rule uses external variables or it uses a parametrized template.

File: linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_login_defs/tests/wrong_value.fail.sh (hunk 3 above)
Comment (matejak) [reply linux_os/guide/system/accounts/accounts-session/user_umask/accounts_umask_etc_login_defs/tests/wrong_value.fail.sh]: This is a good catch, cases like this are hard to handle automatically. I have exempted this rule's tests from the PR.

**Comment by jan-cerny** [comment]
@openscap-ci test this please

**Comment by jan-cerny** [comment]
Hi @matejak!

Thank you.

**Pull request merged**